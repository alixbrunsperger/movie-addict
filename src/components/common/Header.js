import React from 'react';
import Menu from './Menu';
import Search from './Search';
import Logo from './Logo';

export default ({showPopup}) => (
      <div className="header">
        <Logo />
        <Search showPopup={showPopup} />
        <Menu />
      </div>
    );
    
