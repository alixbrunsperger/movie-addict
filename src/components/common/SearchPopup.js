import React from 'react';

export default ({isPopupActive, hidePopup}) => (
      <div className={`search-popup ${isPopupActive ? "active" : ""}`}>
        <div className="search-block">
          <img src="/images/cross.png" onClick={hidePopup} alt="close icon" />
          <form action="/results">
            <input type="text" name="query" />
            <input type="submit" value="Search" />
          </form>
        </div>
      </div>
  );
  
