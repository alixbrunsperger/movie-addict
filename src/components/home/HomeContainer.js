import React, {Component} from 'react';
import Header from '../common/Header';
import SearchPopup from '../common/SearchPopup';
import Home from './Home';

class HomeContainer extends Component {
  state={
    isPopupActive: false
  };

  setIsPopupActive = isPopupActive => this.setState({isPopupActive});

  showPopup = () => this.setIsPopupActive(true)

  hidePopup = () => this.setIsPopupActive(false)

  render() {
    const {isPopupActive} = this.state;
    return (
      <div className="home-container">
        <Header showPopup={this.showPopup}/>
        <SearchPopup isPopupActive={isPopupActive} hidePopup={this.hidePopup} />
        <Home />
      </div>
    );
  }
}
export default HomeContainer;
