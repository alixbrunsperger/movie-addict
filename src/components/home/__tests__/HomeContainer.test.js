import React from 'react';
import { shallow } from 'enzyme';
import HomeContainer from '../HomeContainer';
import Header from '../../common/Header';
import SearchPopup from '../../common/SearchPopup'
describe('HomeContainer', () => {
  it('is rendered', () => {
    const component = shallow(<HomeContainer />);
    expect(component).toMatchSnapshot();
  });

describe('setIsPopupActive', () => {
  it('modify the state when called', () => {
    const component = shallow(<HomeContainer />);
    expect(component.instance().state.isPopupActive).toEqual(false);
    component.instance().setIsPopupActive(true);
    expect(component.instance().state.isPopupActive).toEqual(true);
  });
});

describe('SearchPopup', () => {
  it('pass down isPopupActive from state', () => {
    const component = shallow(<HomeContainer/>);
    const value = "value";
    component.instance().setIsPopupActive(value);
    expect(component.find(SearchPopup).prop("isPopupActive")).toEqual(value);
  });

  it('set hidePopup prop', () => {
    const component = shallow(<HomeContainer/>);
    expect(component.find(SearchPopup).prop("hidePopup")).toEqual(component.instance().hidePopup);
  });
});


describe('Header', () => {
  it('set showPopup prop', () => {
    const component = shallow(<HomeContainer/>);
    const prop = () => component.instance().setIsPopupActive(true);
    expect(component.find(Header).prop("showPopup")).toEqual(component.instance().showPopup);
  });
});
});
