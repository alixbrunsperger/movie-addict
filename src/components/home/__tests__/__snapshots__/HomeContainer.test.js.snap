// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer is rendered 1`] = `
<div
  className="home-container"
>
  <_default
    showPopup={[Function]}
  />
  <_default
    hidePopup={[Function]}
    isPopupActive={false}
  />
  <_default />
</div>
`;
